scripts/ci/pr-bot/processNewPrs.ts

```diff
@@ -0,0 +1,220 @@
+    console.log(
+      `Skipping PR ${pull.number} because it already has been assigned`
+    );
+    return false;
+  }
+  if (prState.stopReviewerNotifications) {
+    console.log(
+      `Skipping PR ${pull.number} because reviewer notifications have been stopped`
+    );
+    return false;
+  }
+
+  return true;
+}
+
+// If the checks passed in via checkstate have completed, notifies the users who have configured notifications.
+async function remindIfChecksCompleted(
+  pull,
+  stateClient: typeof PersistentState,
+  checkState: CheckStatus,
+  prState: typeof Pr
+) {
+  console.log(
+    `Notifying reviewers if checks for PR ${pull.number} have completed, then returning`
+  );
+  if (checkState.completed) {
+    if (checkState.succeeded) {
+      await github.addPrComment(
+        pull.number,
+        commentStrings.allChecksPassed(prState.remindAfterTestsPass)
+      );
+    } else {
+      await github.addPrComment(
+        pull.number,
+        commentStrings.someChecksFailing(prState.remindAfterTestsPass)
+      );
+    }
+    prState.remindAfterTestsPass = [];
+    await stateClient.writePrState(pull.number, prState);
+  }
+}
+
+// If we haven't already, let the author know checks are failing.
+async function notifyChecksFailed(
+  pull,
+  stateClient: typeof PersistentState,
+  prState: typeof Pr
+) {
+  console.log(
+    `Checks are failing for PR ${pull.number}. Commenting if we haven't already and skipping.`
+  );
+  if (!prState.commentedAboutFailingChecks) {
+    await github.addPrComment(
+      pull.number,
+      commentStrings.failingChecksCantAssign()
+    );
+  }
+  prState.commentedAboutFailingChecks = true;
+  await stateClient.writePrState(pull.number, prState);
+}
+
+// Performs all the business logic of processing a new pull request, including:
+// 1) Checking if it needs processed
+// 2) Reminding reviewers if checks have completed (if they've subscribed to that)
+// 3) Picking/assigning reviewers
+// 4) Adding "Next Action: Reviewers label"
+// 5) Storing the state of the pull request/reviewers in a dedicated branch.
+async function processPull(
+  pull,
+  reviewerConfig: typeof ReviewerConfig,
+  stateClient: typeof PersistentState
+) {
+  let prState = await stateClient.getPrState(pull.number);
+  if (!needsProcessed(pull, prState)) {
+    return;
+  }
+
+  let checkState = await getChecksStatus(REPO_OWNER, REPO, pull.head.sha);
+
+  if (prState.remindAfterTestsPass && prState.remindAfterTestsPass.length > 0) {
+    return await remindIfChecksCompleted(
+      pull,
+      stateClient,
+      checkState,
+      prState
+    );
+  }
+
+  if (!checkState.succeeded) {
+    return await notifyChecksFailed(pull, stateClient, prState);
+  }
+  prState.commentedAboutFailingChecks = false;
+
+  // Pick reviewers to assign. Store them in reviewerStateToUpdate and update the prState object with those reviewers (and their associated labels)
+  let reviewerStateToUpdate = {};
```

Comment:
Missing type. This will not transpile under typescript [strict mode](https://www.byte.ski/blog/article/what-is-strict-mode-in-typescript-and-why-and-when-you-should-use-it).